pkg/services/pluginsintegration/clientmiddleware/forward_id_middleware.go

```diff
@@ -0,0 +1,100 @@
+package clientmiddleware
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/services/contexthandler"
+)
+
+const forwardIDHeaderName = "X-Grafana-Id"
+
+// NewForwardIDMiddleware creates a new plugins.ClientMiddleware that will
+// set grafana id header on outgoing plugins.Client requests if the
+// feature toggle FlagIdForwarding is enabled
+func NewForwardIDMiddleware() plugins.ClientMiddleware {
+	return plugins.ClientMiddlewareFunc(func(next plugins.Client) plugins.Client {
+		return &ForwardIDMiddleware{
+			next: next,
+		}
+	})
+}
+
+type ForwardIDMiddleware struct {
+	next plugins.Client
+}
+
+type requestWithHeader interface {
```

Comment:
Nit. Is this really needed. Can't you use the [backend.ForwardHTTPHeaders](https://github.com/grafana/grafana-plugin-sdk-go/blob/4aae405d0a88b4bca46e7e3276a535cb8b3d4236/backend/http_headers.go#L27C6-L27C24) interface?

Reply:
Fixed in [fef8dd4](https://github.com/grafana/grafana/pull/75651/commits/fef8dd40e00a024170f58606e3adb841ec935938)